### ISSUE TITLE
RDTL: "No props found for this component" when the component name is different from its filename

### ISSUE THREAD

**Issue by Vanuan** [storybookjs/storybook#10278]
**Describe the bug**
When there's multiple components per file, only one of their prop tables works.

**To Reproduce**

component file:
```
  import React from 'react';
  
  import { IAProps, IBProps } from './types';
  
  /**
   * A component
   */
  const A = (props: IAProps): JSX.Element => {
    return <>Hi {props.aProperty}</>;
  };
  
  /**
   * B component
   */
  const B = (props: IBProps): JSX.Element => {
    return <>Hi {props.bProperty}</>;
  };
  
  export { A, B };
```

First story:

```
  import React from 'react';
  
  import { A } from './B';
  
  export default {
    title: 'Atoms|A',
    component: A,
  };
  
  export const StoryA = (): JSX.Element => <A aProperty="A property" />;
```

Second story:
```
  import React from 'react';
  
  import { B } from './B';
  
  export default {
    title: 'Atoms|B',
    component: B,
  };
  
  export const StoryB = (): JSX.Element => <B bProperty="B property" />;
```

Only one of these component stories would have the props table populated in storybook.

In this example, only `B` has it. If I rename `B.tsx` to `A.tsx`, `A` would have it.

**Expected behavior**
I expect result should  not depend on the name of a component file.

**Screenshots**
![image](https://user-images.githubusercontent.com/361985/78094934-702aab80-73de-11ea-8130-5a0eacafa9a2.png)
![image](https://user-images.githubusercontent.com/361985/78095077-bf70dc00-73de-11ea-8734-5601e861818f.png)

**Code snippets**
```
    webpackFinal: async config => {
      config.module.rules.push({
        test: /\.(ts|tsx)$/,
        use: [
          {
            loader: require.resolve('ts-loader'),
            options: {
              configFile: path.resolve(__dirname, './tsconfig.json'),
              transpileOnly: true,
            },
          },
          // Automatics props
          {
            loader: require.resolve('react-docgen-typescript-loader'),
            options: {
              tsconfigPath: path.resolve(__dirname, './tsconfig.json'),
            },
          },
        ],
      });
      config.resolve.extensions.push('.ts', '.tsx');
      return config;
    },

```

**System:**
`5.3.18`

**Additional context**

Another issue with RDTL.

**Comment by shilman** [comment]
FYI this works in react-docgen: https://github.com/storybookjs/storybook/pull/10279

**Comment by shilman** [comment]
A few weeks ago we released zero-config typescript support in 6.0-beta, and then a few days ago upgraded `react-docgen-typescript-loader` to `react-docgen-typescript-plugin`. This will solve many typescript props-related issues, possibly including this one:

- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#zero-config-typescript)
- [Updated docs](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/typescript-config/index.md)

Please try it out!
